File: src/models/organization.rs
use serde::{Serialize, Deserialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Organization {
    created_at: String,
    description: String,
    id: String,
    logo: String,
    name: String,
    support_email: String,
    updated_at: String,
    url: String,
}

File: api/src/data/resources/organization.ts
import type { DATA_EVY_Organization } from "evy-types";
import { organization } from "evy-types/db/schema.generated";
import { validateDataEvyOrganization } from "evy-types/validators";
import { makeCoreResource } from "./coreResource";

export const organizationsResource = makeCoreResource<DATA_EVY_Organization>({
	table: organization,
	validate: validateDataEvyOrganization,
	toUpdateSet: (validated) => ({
		name: validated.name,
		description: validated.description,
		logo: validated.logo,
		url: validated.url,
		support_email: validated.support_email,
		visibility: validated.visibility,
	}),
});
